presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -100,18 +99,13 @@
     TableHandle getAlternativeTableHandle(Session session, TableHandle tableHandle, PartitioningHandle partitioningHandle);
 
     /**
-     * Experimental: if true, the engine will invoke pushdownFilter instead of getLayout.
-     *
-     * This interface can be replaced with a connector optimizer rule once the engine supports these (#12546).
-     */
-    boolean isPushdownFilterSupported(Session session, TableHandle tableHandle);
-
-    /**
-     * Experimental: returns table layout that encapsulates the given filter.
-     *
-     * This interface can be replaced with a connector optimizer rule once the engine supports these (#12546).
+     * Experimental: if true, the engine will invoke getLayout otherwise, getLayout will not be called.
+     * This function remains to ensure backwards compatibility, it needs to be removed.
+     * If filter pushdown is required, use a ConnectorPlanOptimizer in the respective connector in order
+     * to push compute into it's TableScan.
      */
-    PushdownFilterResult pushdownFilter(Session session, TableHandle tableHandle, RowExpression filter);
+    @Deprecated
+    boolean isPredicatePushdownEnabled(Session session, TableHandle tableHandle);
```

Comment:
Based on the description, I'd rename this method to `isLegacyGetLayoutSupported`. What kind of backwards compatibility is needed here?

Reply:
Essentially PickTableLayout needs to know that it should not invoke getLayout for queries in which predicatePushdown is enabled. PickTableLayout logic is currently mutually exclusive with pushdownFilter logic. This is the "backwards compatibility" that needs to be supported.

I do think calling this function "isLegacyGetLayoutSupported" is an apt name. However, when reading PickTableLayout `pushPredicateIntoTableScan` is only called when '! isPredicatePushdownEnabled'. Hence in that context, `isPredicatePushdownEnabled` may make more sense.

Let me know if you still think that the name may be misleading or if we can come up with a better a comment to represent what it actually does.